### DISCUSSION TITLE
Projects using cibuildwheel

### DISCUSSION THREAD

**Discussion by henryiii** [pypa/cibuildwheel#485] [Show and tell]
This is a place to post projects that are using cibuildwheel. Every so often we will bundle the projects listed here and add them to the examples.

**Comment by henryiii** [comment]
RTree:  https://github.com/Toblerity/rtree/pull/163

**Reply by henryiii** [reply]
Interesting, discussions don't seem to cross reference.

**Comment by alfaro96** [comment]
We are pleased to announce that the wheel builder of `scikit-learn` has been successfully moved to `cibuildwheel`: scikit-learn/scikit-learn#17921.

**Reply by henryiii** [reply]
And looks like ARM emulation is now in, too!

**Comment by Czaki** [comment]
`psutil` was migrated to cibuildwheel https://github.com/giampaolo/psutil/issues/1550#issuecomment-748401967

**Reply by henryiii** [reply]
I guess they just don't provide wheels for 2.6 & 3.4. First action file I've seen testing FreeBSD. :) Also one of the few projects I've seen that doesn't pin cibuildwheel.

**Reply by Czaki** [reply]
> Also one of the few projects I've seen that doesn't pin cibuildwheel.

This may be my fault. I help during the integration of cibuildwheel and may forget to pin cibuildwheel on end.

**Reply by henryiii** [reply]
I have an idea for pinning in https://github.com/joerick/cibuildwheel/discussions/492

**Reply by Czaki** [reply]
It looks very nice. Dependabot does not work well with python dependencies pinned in yaml file. But with actions may work

**Comment by henryiii** [comment]
Looks like https://github.com/ets-labs/python-dependency-injector, Travis CI (but maybe [moving to GHA](https://github.com/ets-labs/python-dependency-injector/issues/342)?)

**Comment by henryiii** [comment]
Adding a few:
* [numexpr](https://github.com/pydata/numexpr/pull/368/files)
* [PyTables](https://github.com/PyTables/PyTables/tree/master/.github/workflows)
* [dd-trace-py](https://github.com/DataDog/dd-trace-py/pull/1917) (using alternate arch emulation)
* [Tokenizer](https://github.com/OpenNMT/Tokenizer/pull/213)
* [matrixprofile](https://github.com/matrix-profile-foundation/matrixprofile)

**Comment by henryiii** [comment]
Something to keep an eye on, perhaps? https://github.com/mypyc/mypy_mypyc-wheels/pull/11

**Reply by YannickJadoul** [reply]
PR seems to have been merged :-)

**Reply by henryiii** [reply]
First release of MyPy built using cibuildwheel looks like it's due out tomorrow! :) https://github.com/python/mypy/issues/9926

**Comment by henryiii** [comment]
[deepcharles/ruptures, which has a very comprehensive pyproject.toml](https://github.com/deepcharles/ruptures/blob/master/pyproject.toml) that was just added in a big build-system update. A) very useful for the Cython projects I still manage, and b) makes be very glad I'm pretty much free of Cython and use pybind11 instead. :)

**Comment by Czaki** [comment]
imagecodecs package https://github.com/cgohlke/imagecodecs wheels for linux and macos are build using cibuildwheel in repository: https://github.com/czaki/imagecodecs_build repository. 

It may be interesing as it has over 20 external dependecies in compiled libraries. For speedup this project use custom docker images (because often macos needs much more steeps to get working) 

Interesting thing is usage of `libomp` which not comes from brew so wheel is 10.9. Its also use libraries installed in user space and `install_name_tool` for fix to allow delocate to work.  

it also depends on numpy and build wheels for `pypy` using openblas from brew (#402)

**Comment by guillaumekln** [comment]
Hi there!

We just moved to `cibuildwheel` in https://github.com/OpenNMT/CTranslate2. We build wheels for Linux and macOS that include libraries from the [oneAPI toolkit](https://software.intel.com/content/www/us/en/develop/tools/oneapi/base-toolkit.html). The Linux wheels also include CUDA libraries for GPU execution.

`cibuildwheel` has proved to be very useful to add macOS support in addition to Linux. Thanks for your work!

**Comment by henryiii** [comment]
Looks like markupsafe (from pallets project, think used by flask and jinja) uses cibuildwheel: https://github.com/pallets/markupsafe/blob/master/.github/workflows/build.yaml

**Comment by henryiii** [comment]
https://github.com/dib-lab/sourmash - includes emulated Linux ARM builds. :)

**Comment by Czaki** [comment]
I push python-sanppy to use cibuildwheel and its use all Linux emulation now:
https://github.com/andrix/python-snappy/pull/103#issuecomment-780687494

**Comment by maxbachmann** [comment]
I use cibuildwheel in my project https://github.com/maxbachmann/RapidFuzz as well to build wheels for all platform/architecture/python combinations starting Python 3.5

**Comment by henryiii** [comment]
Looks like CMS (the CERN experiment) is using cibuildwheel in a few places now. :) Not sure there's a published package that we could add to our list (CC @nsmith- ?). Here's the original comment: "cibuildwheel taught me things I didn’t want to know, like that i686 platforms have different floating point math behavior", see https://github.com/cms-nanoAOD/correctionlib/blob/9c8d10f4d9330a612b4e7872615b38db012434f3/tests/test_core.py#L124-L127 and https://github.com/cms-nanoAOD/correctionlib/pull/48/commits/f8526886e67ccf8b55e641415d196322429c7a38

**Reply by henryiii** [reply]
Looks like it's here: https://pypi.org/project/correctionlib/

**Reply by nsmith-** [reply]
👋 if anyone can tell me why those tests only fail on i686 containers I'd buy them a 🍺. In practice its not a huge concern.

**Reply by Czaki** [reply]
I'm not sure which float use your code. but there may be difference with precision of representation (64 vs 32 bits)

**Reply by joaander** [reply]
I don't know if this is related to your test failures, but the FPUs on x86 32-bit processors use an internal 80-bit floating point representation, and round down to 64-bit for double or 32-bit for float. When building for x86_64, most compilers (i.e. gcc) default to using SSE instructions which maintain 64-bit precision throughout double precision calculations (or 32-bit for float). The differences in rounding for intermediate values can lead to different defaults.

**Reply by nsmith-** [reply]
I consistently use double throughout. In a few spot checks it seems that the python math stays consistent between platforms but my ast evaluator does not. Its possible my parsing grammar produces a different order of operations that somehow only causes enough rounding errors on i686. I'm not sure.

**Comment by joaander** [comment]
I just started using **cibuildwheel** for one of our projects: https://github.com/glotzerlab/gsd . GSD uses cython and numpy. GSD builds on 64-bit architectures and runs unit tests on the built wheels. 

The conversion process went very well thanks to the excellent documentation for *cibuildwheel*! In the future, our group will look into using **cibuildwheel** for other projects.

**Comment by Zuzu-Typ** [comment]
I've been using cibuildwheel for [PyGLM](https://github.com/Zuzu-Typ/PyGLM) for quite some time now and it has taken a _lot_ of work off my shoulders. 

I used to do all the building "by hand" before, including firing up a Linux VM, just to be able to supply prebuilds. Now **all I need to do is publish a new version and cibuildwheel does everything for me**. It not only builds the wheels for Windows, Linux and Mac (which I didn't even support before) but also automagically uploads them to GitHub and the PyPI!

An amazing tool to say the least. Couldn't be more glad to have it.

**Reply by Czaki** [reply]
Why there are no wheels for python 3.9 for linux? Maybe you should switch to manylinux2010 image?

**Reply by henryiii** [reply]
manylinux1 will not get 3.10, but it does have 3.9. Though, if you use NumPy, then they have stopped building manylinux1 Python 3.9 wheels. I'd recommend using 2010 for 3.9, as any version of pip should support manylinux2010. (Technically, pip on 3.7+ should support manylinux2010, but due to packaging mutalations, Ubuntu 18.04 can get Python 3.8 + Pip 9).

**Reply by Zuzu-Typ** [reply]
> Why there are no wheels for python 3.9 for linux?

I've been wondering that myself. I thought it was simply not supported yet. Thanks for the tip (: 

Using manylinux2010 would mean that python <3.7 won't be built though, right?
Not that it would really matter since those versions are very outdated anyways, but looking at the pypi statistics, a good 10% of downloads still happen to be for the 3.6 version.

**Reply by Czaki** [reply]
You could use two Linux jobs. One with manylinux1 for python 3.5 and 3.6  and one with manylinux2010 for other ones.

**Reply by henryiii** [reply]
No, all versions of manylinux support 3.5-3.9 (the version of manylinux2010 we pin also support 2.7, and manylinux1 supports 2.7). The problem is that newer manylinuxes require newer pip's; CentOS 7 and Ubuntu 18.04 have Python 3.6 + Pip 9, so those systems will not be able to install manylinux2010 wheels. You can also get Python 3.8 on Ubuntu 18.04, but it still has Pip 9, which does not officially support Python 3.8. 😠 

You can see plots here: https://github.com/pypa/manylinux/issues/994#issuecomment-792210874

**Reply by henryiii** [reply]
(Splitting at Python 3.6 is not a bad idea though - _most_ of the time users should have a new enough pip in 3.7+, and that way, when manylinux1 is dropped at the end of the year, it will nicely coincide with the EOL for Python 3.6).

**Reply by Zuzu-Typ** [reply]
Hm.. I've tried using manylinux2010 now and still python 3.9 is not built, so the root of the problem must lie elsewhere.

Interesting side note though, in the log I've found this: _You requested manylinux2010_x86_64 but I have found this wheel is eligible for manylinux1_x86_64._
And sure enough, it supplied both manylinux1 and manylinux2010 wheels, so there is no need to do two jobs.

Do you have any other idea why cp 3.9 might not be built on Linux by any chance?

**Reply by mayeut** [reply]
> Do you have any other idea why cp 3.9 might not be built on Linux by any chance?

@Zuzu-Typ, it seems you're using python 3.5 to install cibuildwheel on linux (from https://travis-ci.com/github/Zuzu-Typ/PyGLM/jobs/496533434). cibuildwheel itself requires python 3.6+  to run since 1.4.2. Using python 3.5, pip installed the last supported version which is 1.4.1 as can be seen in the logs: `cibuildwheel version 1.4.1`.
Initial support for CPython 3.9 was introduced in 1.6.0.
I'd recommend moving to focal distro & python 3.9 as the linux "driver" of cibuildwheel.

**Reply by henryiii** [reply]
I'd also recommend pinning cibuildwheel (maybe by using the action), that will indicate if you are not getting the version you think you are getting, and also will keep you from getting upgrade surprises on your release day. :)

Edit: if you are using Travis, no action, but still can and should pin.

**Reply by Zuzu-Typ** [reply]
@mayeut, perfect, thank you!
Works like a charm now ^^

**Comment by henryiii** [comment]
Looks like uber/h3-py provides cibuiltwheels now. :)

Also robotology/idyntree.

**Comment by xuyxu** [comment]
Our research group has been using cibuildwheel to publish the Python package of [deep forest](https://github.com/LAMDA-NJU/Deep-Forest).

[**Repo**](https://github.com/LAMDA-NJU/Deep-Forest) | [**PyPI**](https://pypi.org/project/deep-forest/) | [**CI**](https://github.com/LAMDA-NJU/Deep-Forest/blob/master/.github/workflows/build-wheels.yml)

Thanks for developing this wonderful tool ;-)

**Comment by MarshalX** [comment]
Iam using cibuildwheel in [tgcalls](https://github.com/MarshalX/tgcalls) to build wheels for macOS (intel) and Windows (x32, x64). My library it’s a binding via pybind11 to c++ library that uses webrtc. So in my GitHub actions workflow there is many 3dparty deps like OpenSSL, ffmpeg, webrtc and so on. I think it’s useful example for other devs.

My PR with adding this project: https://github.com/pypa/cibuildwheel/pull/800

**Comment by bigcat88** [comment]
[Wrapper of libheif](https://github.com/bigcat88/pillow_heif)
x86_64+ aarch64 - MacOS,Linux binary wheels + sdist.

Using CFFI, build libheif, libde265, x265 libs from source.
Creating github release and publish on pypi.

[pypi page](https://pypi.org/project/pillow-heif/#description)

**Comment by henryiii** [comment]
From a quick [search](https://sourcegraph.com/search?q=CIBW_BUILD+count:2000), I found a few more that have around 2k stars or more (was going for 1k+, but there were too many - picked a few fun ones at the end a bit under 1k):

* tornadoweb/tornado (travis)
* pytorch/fairseq
* MagicStack/uvloop
* MagicStack/asyncpg
* samuelcolvin/pydantic
* vaexio/vaex
* cvxpy/cvxpy
* duckdb/duckdb (travis)
* open/triton
* vispy/vispy
* confluentinc/confluent-kafka-python › tools/wheels/build-wheels.bat ?
* tinyobjloader/tinyobjloader (Asure)
* indygreg/PyOxidizer
* deepmind/open_spiel
* google/benchmark
* apache/beam
* Legrandin/pycryptodome
* yaml/pyyaml
* pyutils/line_profiler
* PixarAnimationStudios/OpenTimelineIO
* AcademySoftwareFoundation/OpenColorIO

Total there are over 1K matches for CIBW_BUILD, though that's total matches not repositories that had a match.

**Reply by henryiii** [reply]
https://sourcegraph.com/search?q=context:global+CIBW_BUILD+count:2000+select:repo&patternType=literal&case=yes shows 393 repos. (Edit: now over 600, and we've passed 1.1K stars)

**Comment by henryiii** [comment]
This one is really cool, IMO: https://github.com/ssciwr/clang-format-wheel

See https://twitter.com/HenrySchreiner3/status/1442877941910802433?s=20

**Comment by dvarrazzo** [comment]
psycopg is using cibuildwheel from release 3.0 for Linux, macOS, WIndows, and, from 3.0.3, for Musl Linux too. Thank you very much!

**Comment by TonyXiang8787** [comment]
Hi, our recently published project `power-grid-model` also uses this tool to build. 

https://github.com/alliander-opensource/power-grid-model

- Short description: Python/C++ library for distribution power system analysis
- Platform: Linux-x86_64, windows-amd64, macos-x86_64 and macos-arm64

**Comment by benfred** [comment]
I'm using cibuildwheel with scikit-build to build wheels for https://github.com/benfred/implicit - a library for implicit feedback collaborative filtering. This includes GPU support for linux wheels.

**Comment by MarshalX** [comment]
I am using cibuildwheel with [Python WebRTC](https://github.com/MarshalX/python-webrtc) to build wheels for all platforms. My project is a Python extension that provides bindings to WebRTC M92. It follows the W3C specification with some modifications and additions to make it work better with Python applications and with useful APIs like programmatic audio and video.

**Comment by eriknw** [comment]
I just set up [`cytoolz`](https://github.com/pytoolz/cytoolz) to use cibuildwheel on [Github Actions here](https://github.com/pytoolz/cytoolz/blob/master/.github/workflows/wheels.yml).  This builds _all_ wheels except for win_arm64, which comes out to 83 different wheels.

This creates one wheel per CI job, which I doubt is the preferred way to do it.

**Reply by Czaki** [reply]
> This creates one wheel per CI job, which I doubt is the preferred way to do it.

no. The default way is to create one job per system with a possible split for architecture. Especially on linux there is a used docker container that contains all python versions for a given architecture.

**Reply by henryiii** [reply]
It depends on how long it takes to build each wheel and how many parallel runners you have, and if you have shared before-all setup you can reuse (not common). One job per wheel does maximize parallel runner usage. I usually split up wheels into jobs, but usually have 2-4 wheels per job.

**Reply by eriknw** [reply]
Cool, thanks for replying.  One wheel per job makes it easier to isolate failures, which is the main reason I did this.  It's also plenty fast for me.  It would be easy to change though now that it's working.

If you had to compile an external C library that takes, say, 2 hours to build, what's the best way to split up jobs?  By system and architecture?  What about for Windows and osx?

**Comment by bjlittle** [comment]
We're super stoked to be using `cibuildwheel` to automate the building and testing of binary wheels for our [cf-units](https://github.com/SciTools/cf-units) package on both `macos` and `linux`. See our GitHub Action [ci-wheels.yml](https://github.com/SciTools/cf-units/blob/main/.github/workflows/ci-wheels.yml) for details.

`cf-units` is a Scientific Python package that provides units of measure as required by the [Climate and Forecast (CF) Metadata Conventions](https://cfconventions.org/).

Thanks for such an awesome package! It's made our lives as package maintainers so much easier! 🍻

**Comment by wbarnha** [comment]
`cibuildwheel` is so great! We are finally able to automate wheel builds for [`faust-streaming`](https://github.com/faust-streaming/faust/blob/master/.github/workflows/dist.yml) and its [`python-rocksdb` fork](https://github.com/faust-streaming/python-rocksdb/blob/master/.github/workflows/dist.yml), thank you all so much!

**Comment by JelleBakker** [comment]
[x-IMU3 Software](https://github.com/xioTechnologies/x-IMU3-Software) and [Fusion](https://github.com/xioTechnologies/fusion) is using cibuildwheel for building and testing wheels for Mac, Ubuntu and Windows. It's working great!

**Comment by Marco-Sulla** [comment]
Well, I add also my little project:

https://github.com/Marco-Sulla/python-frozendict/blob/master/.github/workflows/build_primary_wheels.yml
https://github.com/Marco-Sulla/python-frozendict/blob/master/.github/workflows/build_secondary_wheels.yml

**Comment by chernishev** [comment]
Hi! `Desbordante` data profiler uses `cibuildwheel`.

https://github.com/Desbordante/desbordante-core

* Short description: a high-performance data profiler for complex pattern discovery
* Platforms: Linux-x86_64, macos-x86_64 and macos-arm64

Thank you for the tool!

**Comment by henryiii** [comment]
Here's the packages in the top 15,000 on PyPI with a top-level pyproject.toml containing `tool.cibuildwheel`. This doesn't capture environment variables, other TOML files, or separate repos that contain the build.

|  name | rank | Monthly downloads |
|---|---|---|
| numpy | 16 | 433,643,938 |
| pandas | 24 | 331,106,265 |
| aiohttp | 35 | 286,289,440 |
| psutil | 56 | 193,238,892 |
| yarl | 62 | 183,417,881 |
| multidict | 65 | 176,484,666 |
| frozenlist | 74 | 161,377,052 |
| propcache | 82 | 151,142,031 |
| pillow | 85 | 147,114,500 |
| scipy | 89 | 142,815,938 |
| lxml | 105 | 129,461,120 |
| snowflake-connector-python | 113 | 120,559,090 |
| pyzmq | 127 | 109,194,896 |
| black | 215 | 63,929,896 |
| contourpy | 225 | 62,065,191 |
| websockets | 231 | 60,353,363 |
| tornado | 245 | 57,872,055 |
| dulwich | 248 | 56,707,248 |
| pymongo | 282 | 47,844,455 |
| shapely | 290 | 46,565,336 |
| tiktoken | 351 | 35,978,135 |
| uvloop | 375 | 31,158,647 |
| fastavro | 390 | 29,662,845 |
| libcst | 428 | 26,890,807 |
| h5py | 440 | 25,905,647 |
| duckdb | 527 | 20,988,962 |
| blis | 604 | 17,491,882 |
| srsly | 636 | 16,292,317 |
| ninja | 643 | 16,164,588 |
| cymem | 653 | 15,992,096 |
| murmurhash | 656 | 15,942,727 |
| asyncpg | 686 | 15,006,976 |
| cmake | 808 | 10,931,698 |
| frozendict | 844 | 9,801,008 |
| tree-sitter | 887 | 8,695,702 |
| pemja | 889 | 8,682,633 |
| apache-flink | 890 | 8,678,466 |
| ciso8601 | 912 | 8,345,101 |
| shap | 965 | 7,521,953 |
| pycares | 976 | 7,303,147 |
| faiss-cpu | 1015 | 6,848,627 |
| xmlsec | 1052 | 6,414,148 |
| optree | 1090 | 6,039,915 |
| numexpr | 1188 | 5,061,736 |
| pyogrio | 1314 | 4,234,955 |
| memray | 1333 | 4,149,643 |
| uamqp | 1343 | 4,118,605 |
| patchelf | 1434 | 3,712,217 |
| cytoolz | 1490 | 3,472,970 |
| backports-datetime-fromisoformat | 1494 | 3,429,237 |
| rustworkx | 1537 | 3,217,370 |
| sspilib | 1572 | 3,093,393 |
| pikepdf | 1606 | 2,978,592 |
| temporalio | 1654 | 2,811,558 |
| pygit2 | 1674 | 2,730,753 |
| urwid | 1684 | 2,680,208 |
| soxr | 1717 | 2,538,485 |
| pillow-heif | 1724 | 2,530,103 |
| pycurl | 1745 | 2,488,991 |
| jq | 1750 | 2,478,020 |
| pybcj | 1767 | 2,413,565 |
| lupa | 1770 | 2,408,026 |
| pyppmd | 1771 | 2,405,291 |
| astropy | 1859 | 2,194,253 |
| rtree | 1879 | 2,149,886 |
| inflate64 | 1907 | 2,067,466 |
| simsimd | 1978 | 1,939,298 |
| correctionlib | 1993 | 1,909,887 |
| xgrammar | 2057 | 1,797,741 |
| scs | 2093 | 1,748,996 |
| clang-format | 2149 | 1,643,156 |
| immutables | 2151 | 1,637,861 |
| stringzilla | 2160 | 1,621,832 |
| curl-cffi | 2172 | 1,603,415 |
| pybase64 | 2245 | 1,516,818 |
| djlint | 2258 | 1,499,610 |
| numcodecs | 2291 | 1,467,005 |
| ibm-db | 2294 | 1,463,687 |
| blosc2 | 2305 | 1,451,936 |
| xattr | 2356 | 1,369,891 |
| bitstruct | 2375 | 1,351,310 |
| tree-sitter-python | 2376 | 1,346,708 |
| pyhmmer | 2429 | 1,290,831 |
| pillow-avif-plugin | 2439 | 1,277,923 |
| tables | 2464 | 1,241,762 |
| tree-sitter-javascript | 2537 | 1,159,366 |
| nodejs-wheel-binaries | 2561 | 1,134,111 |
| iminuit | 2660 | 1,040,762 |
| boost-histogram | 2697 | 1,009,537 |
| statsforecast | 2740 | 974,291 |
| yappi | 2741 | 971,695 |
| tree-sitter-yaml | 2875 | 867,170 |
| line-profiler | 2894 | 856,482 |
| mapbox-earcut | 2992 | 787,981 |
| pysam | 3006 | 781,826 |
| coreforecast | 3137 | 711,723 |
| tree-sitter-go | 3192 | 687,545 |
| tree-sitter-xml | 3195 | 686,332 |
| pymatgen | 3197 | 685,227 |
| tree-sitter-rust | 3214 | 677,921 |
| tree-sitter-java | 3240 | 661,025 |
| manifold3d | 3255 | 654,672 |
| vhacdx | 3285 | 643,565 |
| scylla-driver | 3303 | 635,132 |
| falcon | 3305 | 633,532 |
| gnureadline | 3312 | 631,133 |
| tree-sitter-cpp | 3375 | 607,548 |
| tree-sitter-bash | 3396 | 604,107 |
| cchardet | 3436 | 594,200 |
| psygnal | 3484 | 577,755 |
| pylance | 3494 | 574,940 |
| tree-sitter-typescript | 3555 | 552,468 |
| libusb-package | 3589 | 541,131 |
| qiskit | 3604 | 537,395 |
| tree-sitter-html | 3625 | 528,387 |
| tree-sitter-json | 3627 | 528,198 |
| tree-sitter-css | 3636 | 525,998 |
| tree-sitter-sql | 3661 | 518,288 |
| tree-sitter-toml | 3662 | 518,237 |
| tree-sitter-regex | 3663 | 518,025 |
| tree-sitter-markdown | 3667 | 517,348 |
| pi-heif | 3751 | 495,468 |
| pytensor | 3758 | 494,005 |
| livekit | 3792 | 483,318 |
| swig | 3897 | 459,176 |
| faust-cchardet | 3929 | 449,190 |
| spglib | 4128 | 407,910 |
| ada-url | 4137 | 406,507 |
| traits | 4144 | 405,042 |
| pydevd | 4170 | 400,251 |
| pytest-codspeed | 4217 | 387,631 |
| tree-sitter-c-sharp | 4229 | 385,681 |
| xatlas | 4361 | 364,752 |
| qiskit-aer | 4473 | 346,030 |
| tree-sitter-embedded-template | 4497 | 343,511 |
| faust-streaming | 4501 | 342,603 |
| reedsolo | 4551 | 335,077 |
| asammdf | 4624 | 326,217 |
| implicit | 5044 | 293,923 |
| islpy | 5167 | 284,886 |
| codecov-cli | 5393 | 270,733 |
| highspy | 5397 | 270,075 |
| ansible-pylibssh | 5402 | 269,671 |
| chiapos | 5472 | 262,048 |
| pact-python | 5532 | 256,177 |
| chiavdf | 5546 | 254,542 |
| simplejpeg | 5594 | 249,432 |
| tree-sitter-ruby | 5640 | 243,782 |
| fuzzyset2 | 5663 | 241,721 |
| threaded | 5702 | 237,513 |
| spacy-transformers | 5754 | 232,173 |
| datasketches | 5783 | 230,270 |
| awkward-cpp | 6024 | 214,025 |
| tree-sitter-php | 6487 | 204,070 |
| psd-tools | 6505 | 202,496 |
| cmarkgfm | 6529 | 200,442 |
| cityhash | 6538 | 200,072 |
| scrypt | 6774 | 182,785 |
| pymeshfix | 6831 | 179,336 |
| ale-py | 6943 | 172,887 |
| dimod | 6972 | 171,347 |
| asv | 6986 | 170,355 |
| editdistpy | 6991 | 170,066 |
| power-grid-model | 6999 | 169,461 |
| cwltool | 7080 | 164,285 |
| sparse-dot-topn | 7141 | 160,475 |
| schema-salad | 7218 | 156,720 |
| snowflake-connector-python-nightly | 7336 | 149,950 |
| tiledb | 7367 | 148,221 |
| uharfbuzz | 7428 | 145,020 |
| tesserocr | 7517 | 141,370 |
| mercurial | 7627 | 136,030 |
| spacy-pkuseg | 7664 | 134,717 |
| clang-tidy | 7687 | 134,020 |
| imgui-bundle | 7716 | 132,956 |
| mattersim | 7905 | 125,820 |
| pyproject-fmt | 7937 | 124,553 |
| anycrc | 7955 | 123,810 |
| dclab | 8151 | 118,346 |
| matrix-synapse | 8324 | 113,623 |
| qiskit-terra | 8349 | 112,751 |
| qutip | 8403 | 111,347 |
| pyopencl | 8440 | 109,938 |
| hogql-parser | 8482 | 108,715 |
| healpy | 8483 | 108,708 |
| ss-pybind11 | 8623 | 105,098 |
| deciphon-core | 8791 | 100,500 |
| questdb | 8893 | 97,580 |
| pylibmagic | 8914 | 97,087 |
| efel | 8929 | 96,750 |
| pymongoarrow | 8951 | 96,087 |
| tree-sitter-c | 8974 | 95,458 |
| twisted-iocpsupport | 9004 | 94,696 |
| apngasm-python | 9056 | 93,418 |
| astropy-healpix | 9060 | 93,377 |
| nvtx | 9080 | 92,926 |
| opencolorio | 9183 | 90,345 |
| mgrs | 9304 | 87,448 |
| dwave-preprocessing | 9326 | 86,846 |
| gemmi | 9388 | 85,955 |
| cocotb | 9447 | 84,655 |
| glum | 9548 | 82,598 |
| minorminer | 9587 | 82,055 |
| pygame-ce | 9594 | 81,877 |
| tabmat | 9599 | 81,752 |
| dwave-samplers | 9761 | 78,937 |
| astyle | 9779 | 78,526 |
| klayout | 9795 | 78,332 |
| aiotieba | 9873 | 76,931 |
| winrt-runtime | 10242 | 70,276 |
| pybind11-rdp | 10279 | 69,716 |
| numpy-rms | 10365 | 68,513 |
| rabbitizer | 10487 | 66,820 |
| reproject | 10518 | 66,291 |
| fast-histogram | 10633 | 64,943 |
| edsnlp | 10639 | 64,870 |
| spiceypy | 10685 | 64,321 |
| pemjax | 10774 | 63,282 |
| fabio | 10775 | 63,282 |
| python-rtmidi | 10789 | 63,061 |
| pynng | 10796 | 62,962 |
| dwave-optimization | 10838 | 62,275 |
| xinference-client | 10912 | 61,192 |
| s5cmd | 11022 | 59,748 |
| numpy-minmax | 11127 | 58,588 |
| phylotrackpy | 11168 | 58,217 |
| scikit-misc | 11247 | 57,381 |
| mplib | 11270 | 57,056 |
| pyslang | 11284 | 56,910 |
| s3torchconnectorclient | 11286 | 56,875 |
| jump-consistent-hash | 11388 | 55,707 |
| libigl | 11419 | 55,322 |
| xinference | 11421 | 55,287 |
| ansys-mapdl-reader | 11461 | 54,775 |
| quicktions | 11530 | 53,942 |
| bx-python | 11544 | 53,833 |
| openjij | 11548 | 53,767 |
| python-mecab-ko | 11594 | 53,297 |
| mlcommons-loadgen | 11686 | 52,483 |
| time-series-anomaly-detector | 11706 | 52,312 |
| fdasrsf | 11730 | 52,141 |
| xoscar | 11757 | 51,870 |
| cylp | 11791 | 51,646 |
| antspyx | 11845 | 51,192 |
| cyksuid | 11873 | 50,891 |
| siphashc | 11921 | 50,453 |
| jij-cimod | 11965 | 50,141 |
| pyxcp | 12181 | 48,448 |
| foldedtensor | 12185 | 48,420 |
| blspy | 12187 | 48,380 |
| causalml | 12287 | 47,523 |
| pybit7z | 12293 | 47,458 |
| etcpak | 12333 | 47,112 |
| pyrodigal | 12358 | 46,960 |
| python-ics | 12562 | 45,612 |
| libzim | 12744 | 44,147 |
| cutadapt | 12859 | 43,301 |
| tarina | 12920 | 42,890 |
| dnaio | 12979 | 42,361 |
| cornflakes | 13013 | 42,090 |
| monotonic-alignment-search | 13028 | 41,981 |
| flip-evaluator | 13039 | 41,922 |
| clp-ffi-py | 13055 | 41,798 |
| python-stretch | 13082 | 41,651 |
| tract | 13140 | 41,258 |
| uqbar | 13159 | 41,150 |
| win-precise-time | 13289 | 40,233 |
| bulgogi | 13335 | 39,906 |
| cyclonedds-nightly | 13364 | 39,783 |
| python-mbedtls | 13377 | 39,705 |
| kiss-icp | 13378 | 39,700 |
| minecraft-python | 13423 | 39,402 |
| pypop-genomics | 13598 | 38,351 |
| pymimir | 13642 | 38,092 |
| qoi | 13670 | 37,932 |
| fast-simplification | 13770 | 37,418 |
| piqp | 14010 | 36,171 |
| pyemd | 14071 | 35,843 |
| cppyy-cling | 14087 | 35,782 |
| networkx-graph | 14121 | 35,616 |
| mapfile-parser | 14156 | 35,367 |
| spherogram | 14160 | 35,348 |
| imgui | 14171 | 35,307 |
| mqt-core | 14298 | 34,687 |
| whatshap | 14410 | 34,170 |
| ezsnmp | 14431 | 34,062 |
| pocketsphinx | 14601 | 33,428 |
| accelerator-toolbox | 14614 | 33,370 |
| pdfo | 14654 | 33,240 |
| cantera | 14778 | 32,741 |
| aequilibrae | 14794 | 32,650 |
| helics | 14829 | 32,518 |
| grss | 14983 | 31,917 |


And 1949 more not listed above: [remaining.txt](https://github.com/user-attachments/files/20261078/remaining.txt)

CC @tonybaloney.

**Reply by vytas7** [reply]
Thought posting [Falcon](https://falcon.readthedocs.io/en/stable/user/install.html#binary-wheels) but it is already captured automatically in your table :100:

**Edit**: We have just updated to CIBW 3.0 (in the CI, not in any published release yet); it seems to work just fine.

**Comment by fabratu** [comment]
`NetworKit` has been using `cibuildwheel` for quite some time now.

Short description: NetworKit is a toolkit for large-scale network analysis. It aims to provide tools for the analysis of large networks in the size range from thousands to billions of edges.
Platforms: macos arm64/x86_64, linux aarch64,x86_64, windows x86_64

Thanks for the good work :)

**Comment by mara004** [comment]
`pypdfium2` uses cibuildwheel to fill gaps in platform support.
For the most common platforms, Google's build toolchain and sysroots do the job.
But for extra platforms (e.g. `manylinux_{riscv64,loongarch64,s390x}`, anything `musllinux`), we use cibuildwheel with [static clang-cross](https://github.com/mayeut/static-clang-images). Technically, our cibuildwheel config and workflows target all desktop plaforms, but only a subset is used in releases.
